pkg/dns/daemon.go

```diff
@@ -61,71 +62,74 @@ func (d *Daemon) internal(ctx context.Context) (bool, error) {
 		return true, nil
 	}
 
-	logrus.Infof("Renewing and syncing AcornDNS")
+	logrus.Infof("Renewing and syncing AcornDNS...")
 
+	dnsSecret, domain, token, err := d.getDNSSecret(ctx)
+	if err != nil {
+		logrus.Errorf("Failed to get DNS secret: %v", err)
+		return false, nil
+	}
+
+	if err := d.syncIngress(ctx, domain, token, *cfg.AcornDNSEndpoint, dnsSecret); err != nil {
+		logrus.Errorf("Failed to sync ingress: %v", err)
+		return false, nil
+	}
+
+	logrus.Infof("Renewed and synced AcornDNS!")
+
+	return true, nil
+}
+
+func (d *Daemon) getDNSSecret(ctx context.Context) (secret *corev1.Secret, domain string, token string, err error) {
```

Comment:
Typically, named return values are used for very short functions, and in the function you can just set their values and then `return`. In all the return statements in this function, you are currently listing out all four variables. I would recommend either removing the names from the return values, or just setting the variables to what they need to be and `return`ing.